packages/@aws-cdk/assert/test/test.assertions.ts

```diff
@@ -14,6 +14,13 @@ passingExample('expect <stack> at <some path> to have <some type>', () => {
   });
   expect(synthStack).at('/TestResource').to(haveType(resourceType));
 });
+passingExample('expect <stack> at <some path> to have <some type>', () => {
+  const resourceType = 'Test::Resource';
+  const stack = new cdk.Stack();
+  new TestResource(stack, 'TestResource', { type: resourceType });
+
+  expect(stack).at('/TestResource').to(haveType(resourceType));
```

Comment:
I don't think we should need the preceding`/`. I can't think of a single cast where I wouldn't use it.

Reply:
Yeah - but that's the way it is right now, I'm just fixing the behavior so it can work... If we want to change that, then we can make an issue and a subsequent PR :)